build/gen_stub.php

```diff
@@ -377,6 +399,78 @@ public function getArgInfoName(): string {
         }
         return 'arginfo_' . $this->name;
     }
+
+    public function getFlags(): string
+    {
+        $flags = sprintf("ZEND_ACC_%s", strtoupper($this->visibility));
+
+        if ($this->isStatic) {
+            $flags .= "|ZEND_ACC_STATIC";
+        }
+
+        if ($this->isFinal) {
+            $flags .= "|ZEND_ACC_FINAL";
+        }
+
+        if ($this->isAbstract) {
+            $flags .= "|ZEND_ACC_ABSTRACT";
+        }
+
+        if ($this->isDeprecated) {
+            $flags .= "|ZEND_ACC_DEPRECATED";
+        }
+
+        return $flags;
+    }
+
+    public function getDeclarationKey(): string
+    {
+        $name = $this->alias ?? $this->name;
+
+        return "$name|$this->cond";
+    }
+
+    public function hasDeclaration(): bool
+    {
+        return $this->isAbstract === false || $this->visibility !== "public";
```

Comment:
I don't understand this condition. Why is visibility important here?

Reply:
The reason why I did this, is that `ZEND_ABSTRACT_ME` generates the `ZEND_ACC_PUBLIC|ZEND_ACC_ABSTRACT` flags, so I wanted to make sure that `abstract protected` methods won't be included. But I guess then this method name is a bit misleading. What's your suggestion?

Reply:
So there is currently no way to specify an `abstract protected` method using standard macros? In that case I'd suggest throwing an error if `abstract protected` is encountered in stubs.